events/views.py

```diff
@@ -38,3 +41,49 @@ def list_category(request):
         categories = Category.objects.all()
         serializer = CategorySerializer(categories, many=True)
         return Response(serializer.data)
+
+@api_view(['GET'])
+@permission_classes([AllowAny])
+def get_event(request, pk):
+    event = get_object_or_404(Event, pk=pk)
+    serializer = EventSerializer(event)
+    return Response(serializer.data)
+
+@api_view(['POST'])
+@permission_classes([AllowAny])
+def register_for_event(request, pk):
+  
+    serializer = RegistrationSerializer(
+        data={
+            'email': request.data['email'],
+            'event': pk
+        }
+    )
+    serializer.is_valid(raise_exception=True)
+    serializer.save()
+
+    return Response({"message": "Registration created successfully. Email is not verified"})
+
+@api_view(['POST'])
+@permission_classes([AllowAny])
+def verify_otp(request, pk):
+
+    serializer = VerifyOTPSerializer(
+        data={
+            'email': request.data.get('email'),
+            'otp': request.data.get('otp'),
+            'event': pk,
+        }
+    )
+    serializer.is_valid(raise_exception=True)
+
+    registration = get_object_or_404(
+        Registration, 
+        email=serializer.validated_data['email'], 
+        event_id=pk
+    )
+    registration.is_verified = True
+    registration.otp = None
+    registration.save()
```

Comment:
if small set of fields are updated then it's recommended to use update_fields in save as it's more performant